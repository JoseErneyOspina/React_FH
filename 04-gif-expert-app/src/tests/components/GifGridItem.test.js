import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import  GifGridItem  from '../../components/GifGridItem';

describe('Testing at <GifGridItem />', () => {

    const title = 'Un titulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow( <GifGridItem title = { title } url = { url } /> );

    test('should show correctly the component', () => {

        expect( wrapper ).toMatchSnapshot();

    });

    test('shoul have a paragraph with the title', () => {

        const p = wrapper.find( 'p' );
        expect( p.text().trim() ).toBe( title );

    });

    test('should have a image equal to url and alt of the props', () => {

        const img = wrapper.find('img');
        // console.log( img.html() );
        // console.log( img.props().scrc ); // con las props
        // console.log( img.prop( 'src' ) ); // una sola propiedad

        expect( img.prop( 'src' ) ).toBe( url );
        expect( img.prop( 'alt' ) ).toBe( title );

    });

    test('should have a className animate_fadeIn', () => {

        const div = wrapper.find( 'div' );
        //anmate_fadeIn
        // console.log( div.props() );
        // console.log( div.prop( 'className' ) );
        const className = div.prop( 'className' );

        expect( className.includes( 'animate_fadeIn' ) ).toBe( true );

    });

});