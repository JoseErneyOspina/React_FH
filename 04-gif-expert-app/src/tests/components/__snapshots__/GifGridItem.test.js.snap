// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing at <GifGridItem /> should show correctly the component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    Un titulo
     
  </p>
</div>
`;
